cli/src/main.rs

```diff
@@ -33,15 +34,57 @@ pub(crate) enum EncodingOptions {
     },
 }
 
+#[derive(Debug)]
+pub enum FileError {
+    DuplicateError(PathBuf),
+    IoError(std::io::Error)
+}
+
+impl fmt::Display for FileError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            FileError::DuplicateError(file_path) => 
+                write!(f, "Output file {} already exists", file_path.display()),
+            FileError::IoError(io_error) => 
+                write!(f, "{}", io_error),
+        }
+    }
+}
+
+impl From<std::io::Error> for FileError {
+    fn from(err: std::io::Error) -> Self {
+        FileError::IoError(err)
+    }
+}
+
+impl std::error::Error for FileError {}
+
+fn create_file(file_path: &PathBuf) -> Result<File, FileError>{
+    if file_path.exists(){
+        return Err(FileError::DuplicateError(file_path.clone()));
+    }
+    let file = File::create(&file_path)?;
+    Ok(file)
+}
+
 fn main() {
     let opt = Opt::from_args();
     match opt {
         Opt::Decompress { input } => {
             for file_name in input {
+                let mut in_file = File::open(&file_name).unwrap_or_else(|err| {
```

Comment:
In order to remove the duplication which happens due to error handling you could refactor the whole block into a function which returns an appropriate `Result`. Then you can do a single `.expect` or `.unwrap_or_else` on the result (and you could use the `?` operator inside the function.